framework/src/tinyalsa/tinyalsa.c

```diff
@@ -671,6 +671,8 @@ int pcm_readi(struct pcm *pcm, void *data, unsigned int frame_count)
 			} else if (msg.msgId == AUDIO_MSG_XRUN) {
 				/* Underrun to be handled by client */
 				return -EPIPE;
+			} else if (msg.msgId == AUDIO_MSG_COMPLETE) {
```

Comment:
after this time and if one more request comes, we should return specific error, -ESTRPIPE
so below is needed
1. add additional definition in audio.h 
ex) AUDIO_MSG_DISCONNECTED
2. add definition in audio_manager
ex) AUDIO_MANAGER_DEVICE_SUSPENDED 
3. Tinyalsa should handle msg from audio driver here.
4. handle -ESTRPIPE(if it is not in errno.h then we have to add), return AUDIO_MANAGER_DEVICE_SUSPENDED 
5. Media Recorder should handle AUDIO_MANAGER_DEVICE_SUSPENDED.
    If it is returned, stop automatically(do not treat it as a failure)
6. If device is already closed, or some thread already destroyed, return -EDEV
    In this case we can treat it as a fail case.